src/util/transactionLocalStorage.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { TransanctionStatus } from '../types/transactionTypes';
+
+const includeNewTransaction = (transaction: TransanctionStatus) => {
+  const rawTransactions = localStorage.getItem('bridge-ui-transactions');
+  if (!rawTransactions || !rawTransactions.length) {
+    const first = [transaction];
+    return localStorage.setItem('bridge-ui-transactions', JSON.stringify(first));
+  }
+  const transactions = JSON.parse(rawTransactions);
+  transactions.push(transaction);
+  return localStorage.setItem('bridge-ui-transactions', JSON.stringify(transactions));
+};
+
+export { includeNewTransaction };
```

Comment:
The whole logic should be wrapped in `try-catch` in case `localStorage` is not available or the size is exceeded.

Reply:
Note that transactions are also chain-specific so we should have separate `STORAGE_KEYs` for every chain.